service-myplaces/src/main/java/fi/nls/oskari/myplaces/handler/MyPlacesFeaturesHandler.java

```diff
@@ -0,0 +1,209 @@
+import fi.nls.oskari.control.RestActionHandler;
+import fi.nls.oskari.domain.User;
+import fi.nls.oskari.domain.map.MyPlace;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.myplaces.MyPlacesService;
+import fi.nls.oskari.myplaces.service.MyPlacesFeaturesService;
+import fi.nls.oskari.myplaces.service.wfst.MyPlacesFeaturesServiceWFST;
+import fi.nls.oskari.myplaces.service.wfst.MyPlacesFeaturesWFSTRequestBuilder;
+import fi.nls.oskari.service.OskariComponentManager;
+import fi.nls.oskari.service.ServiceException;
+import fi.nls.oskari.util.IOHelper;
+import fi.nls.oskari.util.JSONHelper;
+import fi.nls.oskari.util.ResponseHelper;
+
+@OskariActionRoute("MyPlacesFeatures")
+public class MyPlacesFeaturesHandler extends RestActionHandler {
+
+    private final static Logger LOG = LogFactory.getLogger(MyPlacesFeaturesHandler.class);
+
+    private static final String PARAM_FEATURES = "features";
+    private static final String PARAM_CRS = "crs";
+    private static final String PARAM_LAYER_ID = "layerId";
+    private static final String JSKEY_DELETED = "deleted";
+
+    private MyPlacesService service;
+    private MyPlacesFeaturesService featureService;
+
+    @Override
+    public void init() {
+        super.init();
+        service = OskariComponentManager.getComponentOfType(MyPlacesService.class);
+        featureService = new MyPlacesFeaturesServiceWFST();
+    }
+
+    @Override
+    public void preProcess(ActionParameters params) throws ActionException {
+        params.requireLoggedInUser();
+    }
+
+    @Override
+    public void handleGet(ActionParameters params) throws ActionException {
+        final User user = params.getUser();
+        final String crs = params.getHttpParam(PARAM_CRS, "EPSG:3067");
+        final String layerId = params.getHttpParam(PARAM_LAYER_ID);
+
+        try {
+            final JSONObject featureCollection;
+            if (layerId != null && !layerId.isEmpty()) {
+                LOG.debug("Get MyPlaces by layer id, uuid:", user.getUuid(),
+                        "layerId:", layerId, "crs:", crs);
+                long categoryId = Long.parseLong(layerId);
+                if (!service.canModifyCategory(user, categoryId)) {
+                    throw new ActionDeniedException(
+                            "Tried to GET features from category: " + categoryId);
+                }
+                featureCollection = featureService.getFeaturesByCategoryId(categoryId, crs);
+            } else {
+                LOG.debug("Get MyPlaces by user uuid, uuid:", user.getUuid(),
+                        "crs:", crs);
+                featureCollection = featureService.getFeaturesByUserId(user.getUuid(), crs);
+            }
+
+            ResponseHelper.writeResponse(params, featureCollection);
+        } catch (ServiceException e) {
+            LOG.warn(e);
+            throw new ActionException("Failed to get features");
```

Comment:
Wouldn't it make more sense to attach the cause exception to the ActionException? If you want to log the stacktrace with warning and the message as error you should use ActionParamsException to prevent an unnecessary stacktrace caused by ActionException.